src/rpc/rawtransaction.cpp

```diff
@@ -834,13 +834,9 @@ static UniValue sendrawtransaction(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
     CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
 
-    CFeeRate max_raw_tx_fee_rate = DEFAULT_MAX_RAW_TX_FEE_RATE;
-    // TODO: temporary migration code for old clients. Remove in v0.20
-    if (request.params[1].isBool()) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Second argument must be numeric (maxfeerate) and no longer supports a boolean. To allow a transaction with high fees, set maxfeerate to 0.");
-    } else if (!request.params[1].isNull()) {
-        max_raw_tx_fee_rate = CFeeRate(AmountFromValue(request.params[1]));
-    }
+    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull()
+        ? DEFAULT_MAX_RAW_TX_FEE_RATE
```

Comment:
our clang-format tells me that the question mark goes on the previous line:

```diff
diff --git a/src/rpc/rawtransaction.cpp b/src/rpc/rawtransaction.cpp
index 2f44fd607a..0085954689 100644
--- a/src/rpc/rawtransaction.cpp
+++ b/src/rpc/rawtransaction.cpp
@@ -834,9 +834,9 @@ static UniValue sendrawtransaction(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
     CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
 
-    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull()
-        ? DEFAULT_MAX_RAW_TX_FEE_RATE
-        : CFeeRate(AmountFromValue(request.params[1]));
+    const CFeeRate max_raw_tx_fee_rate{request.params[1].isNull() ?
+                                           DEFAULT_MAX_RAW_TX_FEE_RATE :
+                                           CFeeRate(AmountFromValue(request.params[1]))};
 
     int64_t virtual_size = GetVirtualTransactionSize(*tx);
     CAmount max_raw_tx_fee = max_raw_tx_fee_rate.GetFee(virtual_size);
@@ -906,9 +906,9 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
     CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
     const uint256& tx_hash = tx->GetHash();
 
-    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull()
-        ? DEFAULT_MAX_RAW_TX_FEE_RATE
-        : CFeeRate(AmountFromValue(request.params[1]));
+    const CFeeRate max_raw_tx_fee_rate{request.params[1].isNull() ?
+                                           DEFAULT_MAX_RAW_TX_FEE_RATE :
+                                           CFeeRate(AmountFromValue(request.params[1]))};
 
     CTxMemPool& mempool = EnsureMemPool();
     int64_t virtual_size = GetVirtualTransactionSize(*tx);

Reply:
I reformatted it as above.

However, when I run clang-format it would put everything on one line (145 chars long). This is because `./src/.clang-format` contains `ColumnLimit: 0` (no max line length).

How did you run clang-format so that it would break the line? Have you changed `ColumnLimit` locally from 0 to 90? I will probably do this myself because I don't like the unlimited line length, but it would be cumbersome to drag this local mod all over the place and I wonder if there is something smarter...

Reply:
See https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy 

I run 

```
$ clang-format --version
clang-format version 10.0.0

Reply:
Strange.

```sh
$ git checkout 568cbfa # the commit before I applied the above suggestions
$ git diff -U0 HEAD~1.. | ./contrib/devtools/clang-format-diff.py -p1 -i -v
Formatting src/rpc/rawtransaction.cpp
$ git diff
```

produces (the `max_raw_tx_fee_rate` assignment is all in one, long line):

```diff
diff --git i/src/rpc/rawtransaction.cpp w/src/rpc/rawtransaction.cpp
index 2f44fd607..0fd6597a8 100644
--- i/src/rpc/rawtransaction.cpp
+++ w/src/rpc/rawtransaction.cpp
@@ -821,25 +821,23 @@ static UniValue sendrawtransaction(const JSONRPCRequest& request)
             "\nAs a JSON-RPC call\n"
             + HelpExampleRpc("sendrawtransaction", "\"signedhex\"")
                 },
     }.Check(request);
 
     RPCTypeCheck(request.params, {
-        UniValue::VSTR,
-        UniValueType(), // VNUM or VSTR, checked inside AmountFromValue()
-    });
+                                     UniValue::VSTR,
+                                     UniValueType(), // VNUM or VSTR, checked inside AmountFromValue()
+                                 });
 
     // parse hex string from parameter
     CMutableTransaction mtx;
     if (!DecodeHexTx(mtx, request.params[0].get_str()))
         throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
     CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
 
-    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull()
-        ? DEFAULT_MAX_RAW_TX_FEE_RATE
-        : CFeeRate(AmountFromValue(request.params[1]));
+    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull() ? DEFAULT_MAX_RAW_TX_FEE_RATE : CFeeRate(AmountFromValue(request.params[1]));
 
     int64_t virtual_size = GetVirtualTransactionSize(*tx);
     CAmount max_raw_tx_fee = max_raw_tx_fee_rate.GetFee(virtual_size);
 
     std::string err_string;
     AssertLockNotHeld(cs_main);
@@ -888,30 +886,28 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
             "\nAs a JSON-RPC call\n"
             + HelpExampleRpc("testmempoolaccept", "[\"signedhex\"]")
                 },
     }.Check(request);
 
     RPCTypeCheck(request.params, {
-        UniValue::VARR,
-        UniValueType(), // VNUM or VSTR, checked inside AmountFromValue()
-    });
+                                     UniValue::VARR,
+                                     UniValueType(), // VNUM or VSTR, checked inside AmountFromValue()
+                                 });
 
     if (request.params[0].get_array().size() != 1) {
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Array must contain exactly one raw transaction for now");
     }
 
     CMutableTransaction mtx;
     if (!DecodeHexTx(mtx, request.params[0].get_array()[0].get_str())) {
         throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
     }
     CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
     const uint256& tx_hash = tx->GetHash();
 
-    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull()
-        ? DEFAULT_MAX_RAW_TX_FEE_RATE
-        : CFeeRate(AmountFromValue(request.params[1]));
+    const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull() ? DEFAULT_MAX_RAW_TX_FEE_RATE : CFeeRate(AmountFromValue(request.params[1]));
 
     CTxMemPool& mempool = EnsureMemPool();
     int64_t virtual_size = GetVirtualTransactionSize(*tx);
     CAmount max_raw_tx_fee = max_raw_tx_fee_rate.GetFee(virtual_size);
 
     UniValue result(UniValue::VARR);
```

Reply:
Jup, if the `?` is on a new line to start with, all lines will get compressed. Though, if the `?` is right before a newline character, they won't get compressed. Idk why, just clang-format being clang-format.